File: lib/data/remote/models/article/articles.dart
import 'package:hive/hive.dart';

part 'articles.g.dart';

@HiveType(typeId: 1)
class Article {
  @HiveField(0)
  int id;
  @HiveField(1)
  String title;
  @HiveField(2)
  String date;
  @HiveField(3)
  String minRead;
  @HiveField(4)
  String imageUrl;
  @HiveField(5)
  bool isBookmarked = false;
  @HiveField(6)
  String description;
  @HiveField(7)
  Author author;
  @HiveField(8)
  DateTime createdAt;
  @HiveField(9)
  DateTime addedAt;

  Article(this.id, this.title, this.date, this.minRead, this.imageUrl,
      this.isBookmarked, this.description, this.author, this.createdAt, this.addedAt);
}

@HiveType(typeId: 2)
class Author {
  @HiveField(0)
  String name;
  @HiveField(1)
  String avatar;

  Author(this.name, this.avatar);
}

File: services/articles.js
const db = require('./db');
const helper = require('../helper');
const config = require('../config');
const OneSignal = require('onesignal-node');
var notificationMessage = ""
async function getMultiple(page = 1){
  const offset = helper.getOffset(page, config.listPerPage);
  const rows = await db.query(
    `SELECT * FROM articles LIMIT ?,?`,
    [offset, config.listPerPage]
  );
  const data = helper.emptyOrRows(rows);
  const meta = {page};

  return {
    data,
    meta
  }
}

async function create(article){
  const result = await db.query(
    `INSERT INTO articles
    (title, imageUrl, author, avatar, minRead, description)
    VALUES
    (?, ?, ?, ?, ?, ?)`,
    [
      article.title, article.imageUrl,
      article.author, article.avatar,
      article.minRead, article.description,
    ]
  );

  let message = 'Error in creating articles';
  var notificationMessage = {
    app_id: config.notification.apiKey,
    contents: {"en": article.title},
    included_segments: ["Subscribed Users"],
    small_icon: "app_logo"
  };
  sendNotification(notificationMessage);
  if (result.affectedRows) {
    message = 'Article created successfully';
  }

  return {message};
}

var sendNotification = function(data) {
  var headers = {
    "Content-Type": "application/json; charset=utf-8",
    "Authorization": "Basic " + config.notification.restKey
  };

  var options = {
    host: "onesignal.com",
    port: 443,
    path: "/api/v1/notifications",
    method: "POST",
    headers: headers
  };

  var https = require('https');
  var req = https.request(options, function(res) {
    res.on('data', function(data) {
      console.log("Response:");
      console.log(JSON.parse(data));
    });
  });

  req.on('error', function(e) {
    console.log("ERROR:");
    console.log(e);
  });

  req.write(JSON.stringify(data));
  req.end();
};


module.exports = {
  getMultiple,
  create
}